src/stratigraphy/util/extract_text.py

```diff
@@ -44,3 +44,47 @@ def extract_text_lines(page: fitz.Page) -> list[TextLine]:
             current_line_words = []
 
     return lines
+
+
+def extract_text_lines_from_bbox(page: fitz.Page, bbox: fitz.Rect) -> list[TextLine]:
```

Comment:
Could the extract_text_lines method call this method (using the full page rect as bbox parameter) to avoid duplicated code?